### ISSUE TITLE
Typescript 4.8.2 breaking changes

### ISSUE THREAD

**Issue by alexandre-combemorel** [Quramy/ts-graphql-plugin#917]
**Describe the bug**
It seems that there is a breaking change on typescript 4.8.2, here is the error message I get:
`TypeError: Cannot create property 'pos' on string 'QueryResults'`

I downgraded to version `4.7.4` and I didn't have any issue.

**To Reproduce**
- update typescript version to `4.8.2`
- run `yarn typegen`

**Expected behavior**
`typegen` commande should be running as expected

**debug logs:**
here is the full log I get:
```bash
TypeError: Cannot create property 'pos' on string 'QueryResults'
at setTextRangePos (/Users/user1/workdev/talent-front/community-app/node_modules/typescript/lib/typescript.js:21409:19)
    at Object.setTextRangePosEnd (/Users/user1/workdev/talent-front/community-app/node_modules/typescript/lib/typescript.js:21427:32)
    at createNodeArray (/Users/user1/workdev/talent-front/community-app/node_modules/typescript/lib/typescript.js:23022:16)
    at asNodeArray (/Users/user1/workdev/talent-front/community-app/node_modules/typescript/lib/typescript.js:27553:28)
    at createBaseGenericNamedDeclaration (/Users/user1/workdev/talent-front/community-app/node_modules/typescript/lib/typescript.js:23074:35)
    at Object.createTypeAliasDeclaration (/Users/user1/workdev/talent-front/community-app/node_modules/typescript/lib/typescript.js:25496:24)
    at Object.createTypeAliasDeclaration (/Users/user1/workdev/talent-front/community-app/node_modules/typescript/lib/typescript.js:3324:29)
    at TypeGenVisitor._createTsTypeDeclaration (/Users/user1/workdev/talent-front/community-app/node_modules/ts-graphql-plugin/lib/typegen/type-gen-visitor.js:338:37)
    at Object.leave (/Users/user1/workdev/talent-front/community-app/node_modules/ts-graphql-plugin/lib/typegen/type-gen-visitor.js:105:47)
    at visit (/Users/user1/workdev/talent-front/community-app/node_modules/graphql/language/visitor.js:197:21)```

**Comment by alexandre-combemorel** [comment]
Up

**Comment by Quramy** [comment]
@alexandre-combemorel 
Sorry for my late reply. I fixed this and publish ts-graphql-plugin v3.0.0 .
